mis_builder/models/kpimatrix.py

```diff
@@ -462,23 +464,27 @@ def iter_subcols(self):
             for subcol in col.iter_subcols():
                 yield subcol
 
-    def _load_account_names(self):
-        account_ids = set()
+    def _load_rdi_names(self):
+        rdi_ids = set()
         for detail_rows in self._detail_rows.values():
-            account_ids.update(detail_rows.keys())
-        accounts = self._account_model.search([("id", "in", list(account_ids))])
-        self._account_names = {a.id: self._get_account_name(a) for a in accounts}
-
-    def _get_account_name(self, account):
-        result = "{} {}".format(account.code, account.name)
-        if self._multi_company:
-            result = "{} [{}]".format(result, account.company_id.name)
+            rdi_ids.update(detail_rows.keys())
+        rdi_ids = list(rdi_ids)
+        if UNCLASSIFIED_ROW_DETAIL in rdi_ids:
+            rdi_ids.remove(UNCLASSIFIED_ROW_DETAIL)
+        rdis = self._rdi_model.search([("id", "in", rdi_ids)])
+        self._rdi_names = {rdi.id: self._get_rdi_name(rdi) for rdi in rdis}
+        self._rdi_names[UNCLASSIFIED_ROW_DETAIL] = _("Other")
+
+    def _get_rdi_name(self, rdi):
+        result = rdi.name_get()[0][1]
+        if self._multi_company and rdi.company_id:
+            result = "{} [{}]".format(result, rdi.company_id.name)
         return result
 
-    def get_account_name(self, account_id):
-        if account_id not in self._account_names:
-            self._load_account_names()
-        return self._account_names[account_id]
+    def get_rdi_name(self, rdi_id):
+        if rdi_id not in self._rdi_names:
+            self._load_rdi_names()
+        return self._rdi_names[rdi_id]
```

Comment:
I have done a local test and with this change we avoid possible errors.
```suggestion
        return self._rdi_names.get(rdi_id, _("Other"))
```

Reply:
@FernandoRomera can you provide an scenario to reproduce the problem ?

Reply:
A possible scenario is when you add the option to group by "product_id" and you are looking at a previous year and as you don't use it anymore you have archived the product to avoid problems, but you still have to be able to see it in previous years' reports.

Reply:
For the rest it works great, we have done a couple of tests and it makes it very easy to generate reports.

Reply:
Ok, so maybe it's better to search for inactive records too in `_load_rdi_names()` ?